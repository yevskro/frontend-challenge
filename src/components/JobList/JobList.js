import React from 'react';

function JobList(props){
    return <ul>{props.children}</ul>
}

export default JobList;