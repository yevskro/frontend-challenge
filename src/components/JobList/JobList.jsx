import React from 'react';
import styled from 'styled-components';
import Filters from '../Filters/Filters';
import Filter from '../Filter/Filter';
import Search from '../Search/Search';
import ClearButton from '../ClearButton/ClearButton';
import JobCard from '../JobCard/JobCard';
import jobData from '../../data.json';

function Jobs() {
  return (
    <StyledJobs role="main">
      <Search>
        <Filters>
          <Filter remove>Frontend</Filter>
          <Filter remove>CSS</Filter>
          <Filter remove>JavaScript</Filter>
        </Filters>
        <ClearButton />
      </Search>
      <List>
        <JobCard data={jobData[0]} />
        <JobCard data={jobData[1]} />
        <JobCard data={jobData[2]} />
      </List>
    </StyledJobs>
  );
}

function List({ children }) {
  return <StyledList>{children}</StyledList>;
}

const StyledList = styled.ul`
  margin-bottom: 150px;
`;

const StyledJobs = styled.section`
  position: absolute;
  z-index: 2;
  left: 5%;
  right: 5%;
  top: 120px;
  background: transparent;
  @media (min-width: 401px) {
    left: 11%;
    right: 11%;
  }
`;

export default Jobs;
