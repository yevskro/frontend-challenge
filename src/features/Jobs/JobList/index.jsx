import React from 'react';
import styled from 'styled-components';

function JobList({ children }) {
  return <StyledJobList>{children}</StyledJobList>;
}

const StyledJobList = styled.ul`
  margin-bottom: 150px;
`;

export default JobList;
