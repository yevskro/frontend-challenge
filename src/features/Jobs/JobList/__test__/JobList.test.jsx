import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import JobList from '..';

test('renders', () => {
  render(
    <JobList>
      <p>test</p>
    </JobList>
  );
  expect(screen.getByText('test')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <JobList>
        <p>test</p>
      </JobList>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
