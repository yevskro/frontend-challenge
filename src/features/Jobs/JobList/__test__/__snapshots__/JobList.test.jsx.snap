// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  margin-bottom: 150px;
}

<ul
  className="c0"
>
  <p>
    test
  </p>
</ul>
`;
